### ISSUE TITLE
JSON formatting option "spaceBeforeSeparator" is ignored

### ISSUE THREAD

**Issue by dmak** [revelc/formatter-maven-plugin#929]
**To Reproduce**

Run `mvn formatter:format` on the attached project [test.zip](https://github.com/user-attachments/files/16428380/test.zip) file. Expected that `spaceBeforeSeparator=false` configuration option in `formatter-json.properties` is honoured.

Other configuration options work just fine.

**Versions (OS, Maven, Java, and others, as appropriate):**
 - Affected version(s) of this project: 2.24.1
 - OS: Windows 10 x64

**Additional information**

Additionally I have run the test [`JsonFormatterTest.testDoFormatFileWithConfig()`]() and at least in my setup it has generated the following JSON `target\testoutput\someFile.json` with spaces before field separator:
```
{
  "glossary" : {
    "GlossDiv" : {
      "GlossList" : {
        "GlossEntry" : {
          "Abbrev" : "ISO 8879:1986",
          "Acronym" : "SGML",
          "GlossDef" : {
            "GlossSeeAlso" : [
              "GML",
              "XML"
            ],
            "para" : "A meta-markup language, used to create markup languages such as DocBook."
          },
          "GlossSee" : "markup",
          "GlossTerm" : "Standard Generalized Markup Language",
          "ID" : "SGML",
          "SortAs" : "SGML"
        }
      },
      "title" : "S"
    },
    "title" : "example glossary"
  }
}
```
which is wrong because `spaceBeforeSeparator=false`:
![image](https://github.com/user-attachments/assets/ecca0f1a-0067-4c46-b03b-b29a33686334)

I have noticed that overridden function `DefaultPrettyPrinter#withSeparators(Separators separators)` is not called, which means that `_objectFieldValueSeparatorWithSpaces` is left with default Jackson value ` : `. This is because some of API has become deprecated in Jackson v2.17.
![image](https://github.com/user-attachments/assets/4b9cc1f4-195f-4b58-92c1-ef95c5c2dcea)

One of the solutions could be to initialize `_objectFieldValueSeparatorWithSpaces` directly in `createInstance()` as in [fix.patch](https://github.com/user-attachments/files/16428758/fix.patch).

**Comment by ctubbsii** [comment]
@dmak Thanks for the detailed explanation. Would you be able to open a PR with your proposed patch for easier review and merging?

**Comment by hazendaz** [comment]
fixed via #930

**Comment by ctubbsii** [comment]
@dmak I can look into doing a release today. There are additional updates to jdt that we need to decide whether to include or skip for this release and there are changes to the parent POM that may also need to be included. However, you should not rely on a release to test the changes... ideally, changes should be tested prior to release. If you're waiting for a release to test this, I suggest trying a local snapshot build first to test, and to provide feedback, so we can use that information to have confidence for releasing.

**Comment by hazendaz** [comment]
I think all the sub modules need a refresh as well.  I've been busy with other items and haven't had time to re-group on what all might be needed.  However, if I'm correct we are 3 iterations of Eclipse behind, so maybe push out all three of those and regroup sub modules for March 2025 release provided no concerns there.

**Comment by ctubbsii** [comment]
> so maybe push out all three of those

What do you think about skipping over the ones we missed? After all, it's not a project goal for this plugin to have a release corresponding to every Eclipse JDT release. That has been the case in the past, only because we tended to release more frequently before. But it's not strictly necessary.

**Comment by hazendaz** [comment]
works for me.

**Comment by ctubbsii** [comment]
> works for me.

Okay, I'll work on a release tomorrow then.

**Comment by dmak** [comment]
> I suggest trying a local snapshot build first to test, and to provide feedback, so we can use that information to have confidence for releasing.

I have tested the latest SNAPSHOT build from current Git and it worked OK for me.

**Comment by ctubbsii** [comment]
Sorry I haven't gotten to the release I said I would do. I have gotten busy with preparing for holidays.

**Comment by dmak** [comment]
Is it a good time now to make a release? A New Year release in all aspects 🎉

**Comment by hazendaz** [comment]
One pull request is outstanding.  Once that is reviewed, project is ready for release.  No eta but a solid guess would be less than a month.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ctubbsii** [comment]
released 2.25.0 today
